flink-core/src/main/java/org/apache/flink/types/RowKind.java

```diff
@@ -38,7 +38,7 @@
 	 * to retract the previous row first. It is useful in cases of a non-idempotent update, i.e., an
 	 * update of a row that is not uniquely identifiable by a key.
 	 */
-	UPDATE_BEFORE,
+	UPDATE_BEFORE("UB", (byte) 1),
```

Comment:
One comment about `UB` and `UA`: Have you thought about just calling them `B` and `A`? I know this might also be confusing but having a single character would look nicer when printed to a console otherwise rows will be like:
```
I[]
UB[]
D[]
```

Reply:
I thought about this. The reason I pick `UB` and `UA` is because Oracle uses `I` `D` `UO` `UN` to represent these 4 kinds. I think the "U" in the "UB" string is more meaningful than just a single "B".

Another idea is padding `+` and `-` to `I` `D` `U` to make them have the same length and still can be distinguished easily. What do you think?

```
I+ | a | 12 | FDF
U- | b | 13 | EFC
U+ | b | 14 | CDE
D- | d | 15 | IKH
```

Reply:
Sounds good to me. It nicely shows whether they add or remove rows.